### ISSUE TITLE
Replace DateTime by DateTimeInterface

### ISSUE THREAD

**Issue by nyamsprod** [spatie/opening-hours#12]
### Standalone code to reproduce the problem
```php
<?php

use Spatie\OpeningHours\Time;

$time = Time::fromDateTime(date_create_immutable('2012-11-06 13:25:59.123456'));
```
### Expected
```php
13:25
```

### Actual Result
```php
PHP Fatal error:  Uncaught TypeError: Argument 1 passed to Spatie\OpeningHours\Time::fromDateTime() must be an instance of DateTime, instance of DateTimeImmutable given, called in xxx
```
```

**Comment by TomasVotruba** [comment]
Could you send PR?

**Comment by freekmurze** [comment]
Yup, we would accept a PR with tests that fixes this.

**Comment by freekmurze** [comment]
Fixed in `v1.0.1`
